runtime/src/accounts_db.rs

```diff
@@ -2064,24 +2064,31 @@ impl AccountsDB {
                 }
                 drop(storage);
                 datapoint_debug!("clean_dead_slots", ("stores", stores.len(), i64));
-                let pubkeys: Vec<Vec<Pubkey>> = {
+                let slot_pubkeys: Vec<(Slot, Vec<Pubkey>)> = {
                     self.thread_pool_clean.install(|| {
                         stores
                             .into_par_iter()
                             .map(|store| {
                                 let accounts = store.accounts.accounts(0);
-                                accounts
-                                    .into_iter()
-                                    .map(|account| account.meta.pubkey)
-                                    .collect::<Vec<Pubkey>>()
+                                (
+                                    store.slot,
+                                    accounts
+                                        .into_iter()
+                                        .map(|account| account.meta.pubkey)
+                                        .collect::<Vec<Pubkey>>(),
+                                )
                             })
                             .collect()
```

Comment:
how about `reduce()`-ing here like this, which will be done in parallel by rayon?:

```rust
let pubkeys: HashSet<(Slot, Pubkey)> = 
   ...
   .reduce(|| HashSet<_>::new(), |mut reduced, pubkeys| reduced.extend(pubkeys)

```

so that we can remove the need to construct `cleaned_slot_keys` later?

Reply:
updated!